lib/utils/error_handler.ts

```diff
@@ -5,10 +5,29 @@ import * as express from "express";
  *
  * @param response The Express Response for the current request
  */
-export function handleErrorAndRespond(response: express.Response): (reason: any) => any {
-  return (reason: any) => {
-    response.status(500).json({
+export function handleErrorAndRespond<T>(response: express.Response): (reason: T) => express.Response {
+  return (reason: T) => {
+    return response.status(500).json({
       error: reason,
     });
   };
 }
+
+/**
+ * Returns an error handler that responds to the current HTTP request.
+ *
+ * @param response The Express Response for the current request
+ */
+export function handleNullableResultAndRespond<T, E>(
```

Comment:
probably it's better to do this with an express middleware, so you don't have to call it from inside the controller function (you can pass the middleware to the routing function).